src/components/instance-form-builder/GeneralInfoForm.jsx

```diff
@@ -166,12 +166,21 @@ class GeneralInfoForm extends React.Component {
                 return;
             }
 
-            this.setState({ isSaving: true });
+            this.props.instance.validateInstanceName(this.props.d2).then(isValid => {
+                if (!isValid.status) {
+                    this.props.snackbar.error(
+                        i18n.t("This URL and Username combination already exists")
+                    );
+                    return;
+                }
 
-            this.props.originalInstance.remove(this.props.d2).then(() => {
-                this.props.instance.save(this.props.d2, this.props.isEdit).then(() => {
-                    this.setState({ isSaving: false });
-                    this.props.history.push("/instance-configurator");
+                this.setState({ isSaving: true });
+
+                this.props.originalInstance.remove(this.props.d2).then(() => {
+                    this.props.instance.save(this.props.d2, this.props.isEdit).then(() => {
```

Comment:
It looks like save had a second argument, but now it has only D2 as first argument.